src/client/Microsoft.Identity.Client/AuthScheme/PoP/PoPAuthenticationScheme.cs

```diff
@@ -30,12 +29,20 @@ internal class PoPAuthenticationScheme : IAuthenticationScheme
         /// Currently the signing credential algorithm is hard-coded to RSA with SHA256. Extensibility should be done
         /// by integrating Wilson's SigningCredentials
         /// </remarks>
-        public PoPAuthenticationScheme(HttpRequestMessage httpRequestMessage, IPoPCryptoProvider popCryptoProvider)
+        public PoPAuthenticationScheme(PopAuthenticationConfiguration popAuthenticationConfiguration)
         {
-            _httpRequestMessage = httpRequestMessage ?? throw new ArgumentNullException(nameof(httpRequestMessage));
-            _popCryptoProvider = popCryptoProvider ?? throw new ArgumentNullException(nameof(popCryptoProvider));
+            if (popAuthenticationConfiguration == null)
+            {
+                throw new ArgumentNullException(nameof(popAuthenticationConfiguration));
+            }
+            if (popAuthenticationConfiguration.PopCryptoProvider == null)
+            { 
+                throw new ArgumentNullException(nameof(popAuthenticationConfiguration.PopCryptoProvider));
+            }
+
+            _popAuthenticationConfiguration = popAuthenticationConfiguration;
 
-            var keyThumbprint = ComputeRsaThumbprint(_popCryptoProvider.CannonicalPublicKeyJwk);
+            var keyThumbprint = ComputeRsaThumbprint(popAuthenticationConfiguration.PopCryptoProvider.CannonicalPublicKeyJwk);
```

Comment:
`ComputeRsaThumbprint` doesn't actually perform any RSA operations, it just does a hash. Please rename (I know it was named by me)